src/pages/events/[slug].astro

```diff
@@ -27,17 +29,40 @@ if (!Astro.isPrerendered) {
   ({ data } = Astro.props);
 }
 
+const start = tryParseDateParts(data.startDate);
+const end = tryParseDateParts(data.endDate);
+
 export const getStaticPaths = createGetStaticPath("events");
 ---
 
-<Layout title={data.title}>
-  <article class="grid-container">
-    <h1 set:text={data.title} />
-    <p set:text={data.format} />
-    <p set:text={data.location} />
-    <a class="usa-button" href={data.registrationUrl}> Register </a>
-    <p set:text={data.startDate} />
-    <p set:text={data.endDate} />
-    <section class="usa-prose" set:text={data.description} />
-  </article>
+<Layout title="events">
```

Comment:
this kind of page doesn't need to check whether these props are available, does it? (description, date, anything optional?)

Reply:
I did add a check for description as I think that's the only optional input from the CMS I didn't wrap in a conditional yet